src/main/java/com/synopsys/integration/alert/channel/azure/boards/web/AzureBoardsCustomEndpoint.java

```diff
@@ -106,28 +117,36 @@ private FieldAccessor createFieldAccessor(FieldModel fieldModel) {
                     .ifPresent(fields::putAll);
             }
         } catch (AlertDatabaseConstraintException ex) {
-            logger.error("Error creating field acessor for Azure authentication", ex);
+            logger.error("Error creating field accessor for Azure authentication", ex);
         }
         return new FieldAccessor(fields);
     }
 
-    private String createAuthURL(String clientId, String alertServerUrl) {
+    private boolean isAuthenticated(FieldAccessor fieldAccessor) {
+        AzureBoardsProperties properties = AzureBoardsProperties.fromFieldAccessor(azureBoardsCredentialDataStoreFactory, azureRedirectUtil.createOAuthRedirectUri(), fieldAccessor);
+        Proxy proxy = proxyManager.createProxy();
+        return properties.hasOAuthCredentials(proxy);
+    }
+
+    private String createAuthURL(String clientId) {
         StringBuilder authUrlBuilder = new StringBuilder(300);
         authUrlBuilder.append(AzureHttpServiceFactory.DEFAULT_AUTHORIZATION_URL);
-        authUrlBuilder.append(createQueryString(clientId, alertServerUrl));
+        authUrlBuilder.append(createQueryString(clientId));
         return authUrlBuilder.toString();
     }
 
-    private String createQueryString(String clientId, String alertServerUrl) {
-        String authorizationUrl = String.format("%s%s", alertServerUrl, AzureOauthCallbackController.AZURE_OAUTH_CALLBACK_PATH);
+    private String createQueryString(String clientId) {
+        String authorizationUrl = azureRedirectUtil.createOAuthRedirectUri();
         StringBuilder queryBuilder = new StringBuilder(250);
         queryBuilder.append("&client_id=");
         queryBuilder.append(clientId);
         //TODO have an object that stores the request keys and purges them after some amount of time.
-        //TODO also store a redirect URL if possible
         queryBuilder.append("&state=");
         queryBuilder.append(createRequestKey());
-        queryBuilder.append("&scope=vso.work%20vso.code_write");
+        queryBuilder.append("&scope=");
+        queryBuilder.append(AzureOAuthScopes.PROJECTS_READ.getScope());
+        queryBuilder.append("%20");
+        queryBuilder.append(AzureOAuthScopes.WORK_FULL.getScope());
         queryBuilder.append("&redirect_uri=");
         queryBuilder.append(URLEncoder.encode(authorizationUrl, StandardCharsets.UTF_8));
```

Comment:
Why not URI Encode the resulting string? That way we don't have the "%20" and other encoding in the mix.